src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -408,19 +408,7 @@ public static bool SaveMapFile (IBuildEngine4 engine, string mapFile, Dictionary
 			var cachedMap = engine?.GetRegisteredTaskObjectAssemblyLocal<Dictionary<string, HashSet<string>>> (mapFile, RegisteredTaskObjectLifetime.Build);
 			if (cachedMap != null)
 				return cachedMap;
-			var map = new Dictionary<string, HashSet<string>> ();
-			if (!File.Exists (mapFile))
-				return map;
-			foreach (var s in File.ReadLines (mapFile)) {
-				var items = s.Split (new char [] { ';' }, count: 2);
-				var key = items [0];
-				var value = items [1];
-				HashSet<string> set;
-				if (!map.TryGetValue (key, out set))
-					map.Add (key, set = new HashSet<string> ());
-				set.Add (value);
-			}
-			return map;
+			return LoadCustomViewMapFile (mapFile);
```

Comment:
Shouldn't we "register" `mapFile`?  (And why didn't we do this before, given that the beginning of this method calls `.GetRegisteredTaskObjectAssemblyLocal<…>(…)`?)

```csharp
var map  = LoadCustomViewMapFile (mapFile);
if (map != null) {
    engine.RegisterTaskObject (mapFile, map, RegisteredTaskObjectLifetime.Build, allowEarlyCollection: false);
}
return map;
```

Reply:
Oh the code changed... ah cos of the linker can't use RegisterTaskObject

Reply:
Sorry just actually read your comment (its early). The Registration of the file happens in `SaveCustomViewMapFile`.

Reply:
Yeah, the new method that doesn't support `RegisterTaskObject` is for the linker/trimmer step assembly. It doesn't have access to MSBuild APIs.